Business.Queries/Similar/GetAllSimilarPositionsByPositionIdQueryHandler.cs

```diff
@@ -44,29 +46,32 @@ public async Task<List<JobPositionDto>> HandleAsync(GetAllSimilarPositionsByPosi
 
             var sameLevelCompetencies = allPositionCompetencyRatings.Where(e => query.SameLevelCompetencyId.Any(sl => sl == e.Key)).ToDictionary(k => k.Key, v => v.Value);
             var higherLevelCompetencies = allPositionCompetencyRatings.Where(e => query.HigherLevelCompetencyId.Any(sl => sl == e.Key)).ToDictionary(k => k.Key, v => v.Value);
+            var sameorhigherLevelCompetencies = allPositionCompetencyRatings.Where(e => query.SameOrHigherLevelCompetencyId.Any(sl => sl == e.Key)).ToDictionary(k => k.Key, v => v.Value);
+            var allCurrentCompetencies = allPositionCompetencyRatings.Keys.ToList();
 
             var resultCertificates = (
                     await _db.JobRolePositionCertificates
                     .Include(e => e.Certificate).ToListAsync()
                 )
                 .GroupBy(e => new { e.JobGroupId, e.JobGroupLevelId, e.JobPositionId })
-                 .Select(g => new
+                .Select(g => new
                  {
                      JobGroupId = g.Key.JobGroupId,
                      JobGroupLevelId = g.Key.JobGroupLevelId,
                      JobPositionId = g.Key.JobPositionId,
                      Certificates = g.ToList()
                  })
                 .Where(e =>
-                    !query.CertificateId.Any() ||
+                    query.CertificateId.Any() &&
                     query.CertificateId.All(sl => e.Certificates.Any(cr => cr.CertificateId == sl))
                 )
                 .Select(e => new JobPositionDto()
                 {
                     JobGroupId = e.JobGroupId,
                     JobGroupLevelId = e.JobGroupLevelId,
                     JobTitleId = e.JobPositionId,
-                });
+                })
+                .Where(e => e.JobTitleId != query.JobPositionId);
```

Comment:
Where statements can be grouped together (either in the same statement or beside each other)
Select statements can be grouped together.